core/indigo-core/common/math/algebra.h

```diff
@@ -822,5 +824,234 @@ namespace indigo
         float _d;
     };
 
+    inline bool isPointInPolygon(const Vec2f& p, const std::vector<Vec2f>& poly)
+    {
+        bool in = false;
```

Comment:
What about next two corner cases?
![image](https://github.com/user-attachments/assets/a91d311a-91d4-4107-a1e2-d4c94757fae0)

Reply:
Same for edge:
![image](https://github.com/user-attachments/assets/9db2407f-028e-4344-b912-80bd83d6c543)

Reply:
fixed